### DISCUSSION TITLE
Extended Functionality

### DISCUSSION THREAD

**Discussion by maxklt** [jgromes/RadioLib#692] [General]
Hi,

is it possible to read the parameters of LoRa.h (message ID, sender address, receiver address) and also write them in the same scheme via RadioLib?

```
LoRa.beginPacket();
LoRa.write(destinationAddress);
LoRa.write(senderAddress);
LoRa.write(msgID);
LoRa.write(outgoing.length());
LoRa.print(outgoing);
LoRa.endPacket();
```

Best
Mex

**Comment by jgromes** [comment]
None of those parameters are specific to SX127x modules and/or LoRa packet, so I suspect it's something implemented internally by the LoRa.h library. RadioLib doesn't enforce any packet format, so whatever you implement is up to you - you can write arbitrary binary data to any of the transmit/receive methods.

**Reply by maxklt** [reply]
Many thanks!
